asl/encoding/encoding.go

```diff
@@ -82,13 +95,22 @@ func ReadUint64FromBufferLE(buffer []byte, offset int) (uint64, int) {
 }
 
 func ReadUint32FromBufferLE(buffer []byte, offset int) (uint32, int) {
+
 	if IsLittleEndian {
 		// nolint: gosec
 		return *(*uint32)(unsafe.Pointer(&buffer[offset])), offset + 4
 	}
 	return littleEndian.Uint32(buffer[offset:]), offset + 4
 }
 
+func ReadUint32FromBufferLEVarInt(buffer []byte, offset int) (uint32, int) {
```

Comment:
Again, this seems unnecessary. There's a binary.Uvarint method that can be used directly to do this. Here's an example of it being used: https://github.com/spirit-labs/tektite/blob/main/kafkaprotocol/api_versions_request.go#L21

Reply:
this is addressed now